src/Orleans.Runtime/Core/ManagementGrain.cs

```diff
@@ -216,6 +216,17 @@ public ValueTask<SiloAddress> GetActivationAddress(IAddressable reference)
         {
             var grainReference = reference as GrainReference;
             var grainId = grainReference.GrainId;
+            var grainProperties = siloManifest.Grains[grainId.Type].Properties;
```

Comment:
Last comment, I promise :)

While I think it's a good idea to check first the local `GrainManifest` first (it only contains the type supported by the local silo), if the current type is not found (in the heterogeneous silo case for example), you should fall back and use the `ClusterManifest.Current`, that will contains all the `GrainManifest` from the cluster.

We could also generally be more defensive and throw a specific exception if the type is not known, even in the  `ClusterManifest.Current`.

Reply:
Sure. Where should I put that new Exception type?

Reply:
No I do a `TryGet` on the `Grains` dictionary and throw an explicit `ArgumentException` or similar, instead of throwing a criptic `KeyNotFoundException`

Reply:
AhHHH ok, that is already changed. Was thinking you wanted a new class for the exception. Thanks!